pkg/providers/amifamily/bootstrap/bootstrap.go

```diff
@@ -123,4 +135,7 @@ func joinParameterArgs[K comparable, V any](name string, m map[K]V, separator st
 // Examples are the Bottlerocket config and the eks-bootstrap script
 type Bootstrapper interface {
 	Script() (string, error)
+
+	// TODO: jmdeal@ remove once KubeletConfiguration can be properly hashed
+	String() string
```

Comment:
Also, I think that you can scope the blockDeviceMappings hashing into this function so that you don't have to move that hash logic elsewhere like you have it right now

Reply:
So I went with `String` to implement the `fmt.Stringer` interface and just hash `Options` as a string in each of the bootstrap implementations. This seemed like the most straightforward way to handle this since it didn't require any changes further up the call stack. I see an argument to try and keep the custom hashing logic all in one place, particularly because it can be ripped out as soon as we're able to make breaking hash changes for NodePool and NodeClass.